Update ResourceDownloader.showPublicPipelines interface and README.md

DevinTDHa opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
- updated `ResourceDownloader.showPublicPipelines` to be more in line with the `ResourceDownloader.showPublicModels` function
- providing no version will use the current version to check for compatibility
- updated README.md to show examples on how to use those functions

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Added tests for `ResourceDownloader.showPublicPipelines`.

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Code improvements with no or little impact
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

maziyarpanahi commented:
@DevinTDHa 
This is great, can we have also Python examples (the import and the function so they can copy/paste)? Also, one example of how to use an annotator's name with version to see all the available models. (same for pipelines)

DevinTDHa commented:
> @DevinTDHa This is great, can we have also Python examples (the import and the function so they can copy/paste)? Also, one example of how to use an annotator's name with version to see all the available models. (same for pipelines)

I have added the examples and fixed the python README.md so it only has python code now.

Additionally, commit 3c82fde fixes some issues, regarding the python interface. It would work in the python test but the the output wouldn't be properly forwarded to notebooks.
